pkg/alertmanager/operator.go

```diff
@@ -1790,3 +1796,25 @@ func ListOptions(name string) metav1.ListOptions {
 func tlsAssetsSecretName(name string) string {
 	return fmt.Sprintf("%s-tls-assets", prefixedName(name))
 }
+
+func GetAppliedConfigFromCondition(c *monitoringv1.Condition) *monitoringv1ac.ConditionApplyConfiguration {
+	return monitoringv1ac.Condition().
+		WithType(c.Type).
+		WithStatus(c.Status).
+		WithLastTransitionTime(c.LastTransitionTime).
+		WithReason(c.Reason).
+		WithMessage(c.Message).
+		WithObservedGeneration(c.ObservedGeneration)
+}
+
+func GetAppliedConfigStatusFromStatus(a *monitoringv1.Alertmanager, asac *monitoringv1ac.AlertmanagerStatusApplyConfiguration, conditions ...monitoringv1.Condition) {
```

Comment:
Thinking more about it, we can have a single function that takes a monitoringv1.Alertmanager object with an up-to-date status field and returns an AlertmanagerApplyConfiguration object. This would minimize the changes in UpdateStatus().

```golang
func ApplyConfigurationFromAlertmanager(
    a *monitoringv1.Alertmanager
) monitoringv1ac.AlertmanagerApplyConfiguration {
    asac := monitoringv1ac.AlertmanagerStatus().
        WithPaused(a.Status.Paused).
        ...

    ...

    return monitoringv1ac.Alertmanager(a.Name, a.Namespace).WithStatus(asac)
}
```

Reply:
done, I think now is simpler and prettier